packages/wrangler/src/api/startDevWorker/ConfigController.ts

```diff
@@ -399,10 +399,10 @@ async function resolveConfig(
 	// for pulling containers, we need to make sure the OpenAPI config for the
 	// container API client is properly set so that we can get the correct permissions
 	// from the cloudchamber API to pull from the repository.
-	const needsPulling = resolved.containers?.some(
-		(c) => !isDockerfile(c.image ?? c.configuration?.image, config.configPath)
+	const needsPulling = resolved.containers.some(
+		(c) => "registry_link" in c && c.registry_link
 	);
-	if (needsPulling && !resolved.dev.remote) {
+	if (resolved.containers.length && needsPulling && !resolved.dev.remote) {
```

Comment:
why this addition?

if `needsPulling` is `true` then there must at least be a container in `resolved.containers` no?